CFC_DataCollector/recommender/common.py

```diff
@@ -0,0 +1,162 @@
+import urllib2
+import xml.etree.ElementTree as ET
+from get_database import *
+from datetime import datetime, timedelta
+import trip
+from trip import E_Mission_Trip
+import random
+import json
+import jsonpickle
+
+def get_uuid_list():
+	uuid_list = [ ]
+	db = get_section_db()
+	for x in db.find():
+   		uuid_list.append(x['_id'])
+	return uuid_list
+
+def insert_into_pdb(pdb, my_id, new_perturbed_trip):
+	my_id = str(my_id)
+	assert(type(my_id) == str)
+	new_id = my_id.replace('.', "")
+	assert(type(new_id) == str)
+	if '.' not in new_id:
+		pdb.insert({'joh' : new_perturbed_trip})
+    
+def initialize_empty_perturbed_trips(_id):
+	# Assuming I have the methods json_to_trip and trip_to_json
+	db = get_section_db()
+	pdb = get_perturbed_trips_db()
+	json_trip = db.find_one({"_id" : _id})
+	new_perturbed_trip = { }
+	trip = E_Mission_Trip(json_trip)
+	for pert in find_perturbed_trips(trip):
+		pert._id = pert._id.replace('.', '') 
+		new_perturbed_trip[pert._id] = None
+	insert_into_pdb(pdb, trip._id, new_perturbed_trip)
+	return
+
+
+def update_perturbations(_id, perturbed_trip):
+	## Assuming I have trip_to_json function
+	db = get_perturbed_trips_db()
+	json_trip = db.find_one({"_id" : _id})
+	json_trip[perturbed_trip._id] = jsonpickle.encode(perturbed_trip)
+	return
+
+#def query_perturbed_trips()
+
+
+def meters_to_miles(meters):
+	return meters * 0.000621371
+
+def calc_car_cost(trip_id, distance):
+	uuid = sectiondb.find_one({'trip_id': trip_id})['user_id']
+	our_user = User.fromUUID(uuid)
+	ave_mpg = our_user.getAvgMpg()
+	gallons =  meters_to_miles(distance) / ave_mpg
+	price = urllib2.urlopen('http://www.fueleconomy.gov/ws/rest/fuelprices')
+	xml = price.read()
+	p = ET.fromstring(xml)[-1]
+	return float(p.text)*gallons
+
+def store_trip_in_db(trip, is_alternate):
+	""" Stores a trip in the databse, specify whether or not the trip is an alternative """
+	db = get_section_db()
+	to_insert = { }
+	to_insert['_id'] = trip._id
+	#to_insert['distance'] = trip.distance
+	to_insert['section_start_datetime'] = trip.start_time
+	to_insert['section_end_datetime'] = trip.end_time
+	to_insert['section_start_point'] = {'type' : 'Point', 'coordinates' : trip.start_point}
+	to_insert['section_end_point'] = {'type' : 'Point', 'coordinates' : trip.end_point}
+	to_insert['cost'] = trip.cost
+	to_insert['single_mode'] = str(single_mode)
+	to_insert['legs'] = [ ]
+	for leg in trip.legs:
+		l = { }
+		l['mode'] = leg.mode
+		l['cost'] = leg.cost
+		l['duration'] = leg.duration
+		l['distance'] = distance
+		l['start_point'] = {'type' : 'Point', 'coordinates' : leg.starting_point}
+		l['end_point'] = {'type' : 'Point', 'coordinates' : leg.ending_point}
+		to_insert['legs'].append(l)
+	if is_alternate:
+		t = db.find_one({'trip_id' : trip.parent_tid})
+		try:
+			t['alternate_trips'].append(to_insert)
+		except:
+			t['alternate_trips'] = []
+			t['alternate_trips'].append(to_insert)
+	else:
+		db.insert(to_insert)
+
+
+def trip_to_json(trip):
```

Comment:
Sorry, For some reason thats older code

I'm just scared to push anything right now

On Sun, Mar 29, 2015 at 9:49 PM, shankari <notifications@github.com> wrote:

> In CFC_DataCollector/recommender/common.py
> <https://github.com/e-mission/e-mission-server/pull/103#discussion_r27366588>
> :
>
> > +		l['distance'] = distance
> > +		l['start_point'] = {'type' : 'Point', 'coordinates' : leg.starting_point}
> > +		l['end_point'] = {'type' : 'Point', 'coordinates' : leg.ending_point}
> > +		to_insert['legs'].append(l)
> > +	if is_alternate:
> > +		t = db.find_one({'trip_id' : trip.parent_tid})
> > +		try:
> > +			t['alternate_trips'].append(to_insert)
> > +		except:
> > +			t['alternate_trips'] = []
> > +			t['alternate_trips'].append(to_insert)
> > +	else:
> > +		db.insert(to_insert)
> > +
> > +
> > +def trip_to_json(trip):
>
> @joshzarrabi <https://github.com/joshzarrabi>, why not use jsonpickle
> here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/e-mission/e-mission-server/pull/103/files#r27366588>.
>